repose-aggregator/components/filters/destination-router-filter/src/main/scala/org/openrepose/filters/destinationrouter/DestinationRouterFilter.scala

```diff
@@ -20,36 +20,34 @@
 package org.openrepose.filters.destinationrouter
 
 import java.util
-import java.util.concurrent.TimeUnit
+import java.util.Optional
 import javax.inject.{Inject, Named}
 import javax.servlet._
 import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
 
+import com.codahale.metrics.{Meter, MetricRegistry}
 import com.typesafe.scalalogging.slf4j.LazyLogging
 import org.openrepose.commons.config.manager.UpdateListener
 import org.openrepose.commons.utils.http.CommonRequestAttributes
 import org.openrepose.commons.utils.servlet.http.RouteDestination
 import org.openrepose.core.filter.FilterConfigHelper
-import org.openrepose.core.filters.DestinationRouter
 import org.openrepose.core.services.config.ConfigurationService
 import org.openrepose.core.services.reporting.metrics.MetricsService
 import org.openrepose.filters.routing.servlet.config.DestinationRouterConfiguration
 
 @Named
-class DestinationRouterFilter @Inject()(configurationService: ConfigurationService, metricsService: MetricsService)
+class DestinationRouterFilter @Inject()(configurationService: ConfigurationService, metricsService: Optional[MetricsService])
   extends Filter with UpdateListener[DestinationRouterConfiguration] with LazyLogging {
 
   private final val DefaultConfigFileName = "destination-router.cfg.xml"
   private final val SchemaFilePath = "/META-INF/schema/config/destination-router-configuration.xsd"
 
-  private val mbcsRoutedReponse = metricsService.newMeterByCategorySum(classOf[DestinationRouter],
-    "destination-router",
-    "Routed Response",
-    TimeUnit.SECONDS)
-
   private var initialized: Boolean = false
   private var configurationFileName: String = _
   private var configuration: DestinationRouterConfiguration = _
+  private var routedResponseMetric: Option[Meter] = None
+  private val metricsServiceOption = Option(metricsService.orElse(null))
+  private val getRoutedResponseMetricName = MetricRegistry.name(classOf[DestinationRouterFilter], "Routed Response", _)
```

Comment:
I'm not sure if this is any better, but Scala allows us to partially apply a function. As a result, we can reduce duplication when it comes to metric names in Scala. In this case, we aren't really reducing duplication, just giving a name to the function that specifically creates a metric name from a target ID.